### PR TITLE
113 function to convert fieldspec field mapping to jqfieldspec field mapping

### PR DESCRIPTION
**Pull request by FreddieMatherSmartDCSIT**
PR to test functionality that converts FieldSpec to JQFieldSpec and validates inputs to check consistency

### CODE REVIEW

File: tel2puml/otel_to_pv/config.py [hunk 1]
```diff
@@ -26,19 +30,264 @@ class SpanSpec(TypedDict):
...
+    key_paths: list[str | Iterable[str]]
+    key_value: Optional[list[Optional[Union[str, Iterable[str | None]]]]]
+    value_paths: Optional[list[Optional[Union[str, Iterable[str | None]]]]]
     value_type: Union[str, int]
 
 
-class JQFieldSpec(TypedDict):
-    """Typed dict for JQFieldSpec."""
+class JQFieldSpec:
+    """Typed dict for JQFieldSpec.
```
Comment (AdrianMontaguSmartDCSIT) [inline tel2puml/otel_to_pv/config.py:40]: Update docstring

File: tel2puml/otel_to_pv/config.py [hunk 2]
```diff
@@ -26,19 +30,264 @@ class SpanSpec(TypedDict):
...
+            priority_key_path = tuple(iter(key_path))
+            for key in priority_key_path:
+                if not isinstance(key, str):
+                    raise ValueError(
+                        "Priority key path, within an iterable, must be a "
+                        "string"
+                    )
+            return priority_key_path
+        except TypeError:
+            raise ValueError("Key path must be iterable or a string")
```
Comment (AdrianMontaguSmartDCSIT) [inline tel2puml/otel_to_pv/config.py:159]: Do we want to raise a TypeError here?

File: tel2puml/otel_to_pv/config.py (hunk 1 above)
Comment (FreddieMatherSmartDCSIT) [reply tel2puml/otel_to_pv/config.py]: updated

File: tel2puml/otel_to_pv/config.py (hunk 2 above)
Comment (FreddieMatherSmartDCSIT) [reply tel2puml/otel_to_pv/config.py]: Yeah thats better idea actually - updated

**Review by AdrianMontaguSmartDCSIT — APPROVED** [review]
Looks good to me!

**Pull request merged**